src/main/java/edu/harvard/iq/dataverse/engine/command/impl/RegisterOIDCUserCommand.java

```diff
@@ -0,0 +1,128 @@
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+import edu.harvard.iq.dataverse.engine.command.exception.IllegalCommandException;
+import edu.harvard.iq.dataverse.engine.command.exception.PermissionException;
+import edu.harvard.iq.dataverse.engine.command.exception.InvalidFieldsCommandException;
+import edu.harvard.iq.dataverse.util.BundleUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RequiredPermissions({})
+public class RegisterOIDCUserCommand extends AbstractVoidCommand {
+
+    private final String bearerToken;
+    private final UserDTO userDTO;
+
+    public RegisterOIDCUserCommand(DataverseRequest aRequest, String bearerToken, UserDTO userDTO) {
+        super(aRequest, (DvObject) null);
+        this.bearerToken = bearerToken;
+        this.userDTO = userDTO;
+    }
+
+    @Override
+    protected void executeImpl(CommandContext ctxt) throws CommandException {
+        try {
+            OIDCUserInfo oidcUserInfo = ctxt.authentication().verifyOIDCBearerTokenAndGetUserIdentifier(bearerToken);
+            UserRecordIdentifier userRecordIdentifier = oidcUserInfo.getUserRecordIdentifier();
+
+            if (ctxt.authentication().lookupUser(userRecordIdentifier) != null) {
+                throw new IllegalCommandException(BundleUtil.getStringFromBundle("registerOidcUserCommand.errors.userAlreadyRegisteredWithToken"), this);
+            }
+
+            UserInfo userClaimsInfo = oidcUserInfo.getUserClaimsInfo();
+
+            // Update the UserDTO object with available OIDC user claims; keep existing values if claims are absent
+            userDTO.setUsername(getValueOrDefault(userClaimsInfo.getPreferredUsername(), userDTO.getUsername()));
+            userDTO.setFirstName(getValueOrDefault(userClaimsInfo.getGivenName(), userDTO.getFirstName()));
+            userDTO.setLastName(getValueOrDefault(userClaimsInfo.getFamilyName(), userDTO.getLastName()));
+            userDTO.setEmailAddress(getValueOrDefault(userClaimsInfo.getEmailAddress(), userDTO.getEmailAddress()));
+
+            AuthenticatedUserDisplayInfo userDisplayInfo = new AuthenticatedUserDisplayInfo(
+                    userDTO.getFirstName(),
+                    userDTO.getLastName(),
+                    userDTO.getEmailAddress(),
+                    userDTO.getAffiliation() != null ? userDTO.getAffiliation() : "",
+                    userDTO.getPosition() != null ? userDTO.getPosition() : ""
+            );
+
+            Map<String, String> fieldErrors = validateUserFields(ctxt);
+            if (!fieldErrors.isEmpty()) {
+                throw new InvalidFieldsCommandException(
+                        BundleUtil.getStringFromBundle("registerOidcUserCommand.errors.invalidFields"),
+                        this,
+                        fieldErrors
+                );
+            }
+
+            ctxt.authentication().createAuthenticatedUser(userRecordIdentifier, userDTO.getUsername(), userDisplayInfo, true);
```

Comment:
Would its make sense to use the `updated_at` claim to somehow know when we need to update our copy of user properties?

Reply:
Your suggestion is interesting, though it introduces some complexity in my opinion. 

We'd need to determine the behavior if new claims are either available or unavailable for use within Dataverse. For example, if a username changes in the IdP but is already taken by another user in Dataverse, we’d face a collision right?. 

Additionally, where would be the ideal place to check for these updates? Should we verify on each request using the token, or at regular intervals? Anyway I see this more as a future iteration than something to figure out and implement now.

Reply:
Yes, this is definitely a future requirement. During login/API requests, an out-of-bound request to the userinfo endpoint is required if the `update_at` property within the JWS is newer, or similarly an alternative process probality time based if the `update_at` is not available. 

And yes, we have the same problems as with account creation, and we won't solve them until we drop unique constraints. This has some implications. For most properties it does not matter, but once we miss updates on the email property from the OIDC server we cannot reach the user. This is a problem we already have ....